### PR TITLE
Playable game

### PR DESCRIPTION
**Pull request by Chroq**
## Description

In this PR, we can put some stone on the board. Player indicator changes accordingly.

Mobile :

image

### CODE REVIEW

File: components/boards/Hexagon.js [hunk 1]
```diff
@@ -1,6 +1,21 @@
...
+import {
+  FIRST_PLAYER_COLOR,
+  SECOND_PLAYER_COLOR,
+  LIGHT_COLOR_100
+} from "../../theme/colors";
+
+import { FIRST_PLAYER_VALUE, SECOND_PLAYER_VALUE } from "./Playboard";
 
 function Hexagon(props) {
+  const value = props.value;
```
Comment (Kmaschta) [inline components/boards/Hexagon.js:11]: JS protip: instead if copying every variable from the props, you can destructure them directly from the arguments:

```js
function Hexagon({ value }) {}
```

It is generally more readable

(not mandatory for the review)

File: components/boards/Playboard.js [hunk 2]
```diff
@@ -17,19 +21,36 @@ function Playboard(props) {
...
+      return;
+    }
+    const updatedGrid = grid.map((hexagon, index) =>
+      id === index ? player : hexagon
+    );
+    setGrid(updatedGrid);
+    setPlayer(player === 1 ? 2 : 1);
+  };
+
+  useEffect(() => {}, [grid]);
```
Comment (Kmaschta) [inline components/boards/Playboard.js:38]: This line doesn't make any instruction, you can remove it

File: components/boards/Playboard.js [hunk 3]
```diff
@@ -17,19 +21,36 @@ function Playboard(props) {
   const hexagonWidth = getHexagonWidth(size);
   const hexagonHeight = getHexagonHeight(size);
 
-  const board = [...Array(size)];
+  const [grid, setGrid] = useState(Array(props.size * props.size).fill(0));
```
Comment (Kmaschta) [inline components/boards/Playboard.js:24]: You are implementing a few board logic here (board initialization, insert pebble, etc)
This logic have to be moved into its own files, with everything related to the board and/or game logic

File: components/boards/Playboard.js [hunk 4]
```diff
@@ -39,22 +60,31 @@ function Playboard(props) {
...
+                value={value}
               />
             );
-          });
-        })}
+          })}
+        </div>
       </div>
+
+      <div className="side">{<Hud player={player} />}</div>
```
Comment (Kmaschta) [inline components/boards/Playboard.js:78]: unnecessary {}

File: cypress/integration/board_page_spec.js [hunk 5]
```diff
@@ -7,9 +7,22 @@ describe("The Board Page", function() {
...
+      .find("svg")
+      .should("have.length", size * size);
+    cy.get("p[name=current-player]").contains("Player 1");
+  });
+
+  it("Can put a stone anywhere as player 1 and player 2 can play after", function() {
+    cy.get("div[name=grid]")
+      .find("svg[name=hexagon_0]")
+      .click();
+    cy.get("p[name=current-player]").contains("Player 2");
```
Comment (Kmaschta) [inline cypress/integration/board_page_spec.js:21]: contains must be followed by a should
contains are selectors
should are assertions

File: components/boards/Playboard.js (hunk 2 above)
Comment (Chroq) [reply components/boards/Playboard.js:38]: Fixed, ty

File: components/boards/Playboard.js (hunk 3 above)
Comment (Chroq) [reply components/boards/Playboard.js:24]: Can I move it in the next PR ? I'll work on the grid, game "engine" etc.

File: components/boards/Playboard.js (hunk 4 above)
Comment (Chroq) [reply components/boards/Playboard.js:78]: Fixed !

File: components/boards/Playboard.js (hunk 3 above)
Comment (Chroq) [reply components/boards/Playboard.js:24]: Ty, I resolve the conversation ! :)

File: cypress/integration/board_page_spec.js (hunk 5 above)
Comment (Chroq) [reply cypress/integration/board_page_spec.js:21]: I have applied the documentation : 

image

I don't understand who is right :)

File: components/boards/Hexagon.js (hunk 1 above)
Comment (Chroq) [reply components/boards/Hexagon.js:11]: Ok, I've tried but I introduce breaking changes with the "onClick" function...
I let this version and I'll try later.

File: cypress/integration/board_page_spec.js (hunk 5 above)
Comment (Kmaschta) [reply cypress/integration/board_page_spec.js:21]: If you're talking about this usage: https://docs.cypress.io/api/commands/contains.html#Usage
This part of the doc is focused on "how to use contains" but it do not explain how to do assertions properly

If you take a look at the "should" documentation, you'll see something similar to my suggestion:
https://docs.cypress.io/api/commands/contains.html#Usage

image

**Pull request merged**